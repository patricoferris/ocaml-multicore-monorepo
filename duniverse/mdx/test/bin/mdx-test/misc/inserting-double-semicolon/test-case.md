`;;` is not mandatory at the end of toplevel phrases but will be inserted
automatically in the corrected file. Also, a warning will be emitted.

```ocaml
# let x = 3
val x : int = 3
# let f = function
  | 0 -> 1
  | n -> n-1
val f : int -> int = <fun>
```
