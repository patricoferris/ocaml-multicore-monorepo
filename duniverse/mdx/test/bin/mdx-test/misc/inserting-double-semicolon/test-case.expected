`;;` is not mandatory at the end of toplevel phrases but will be inserted
automatically in the corrected file. Also, a warning will be emitted.

```ocaml
# let x = 3;;
val x : int = 3
# let f = function
  | 0 -> 1
  | n -> n-1;;
val f : int -> int = <fun>
```
Warning: OCaml toplevel block without trailing ;; detected in file 'test-case.md'.
Non-semicolon terminated phrases are deprecated.
In MDX 3.0 support for toplevel blocks without ;; will be removed.
